ext/test_DataFrames.jl

```diff
@@ -0,0 +1,21 @@
+#  Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+module TestContainersDataFrames
+
+using Test
+
+using DataFrames
+using JuMP
+
+function test_dimension_data_variable()
+    model = Model()
+    @variable(model, x[i = 2:4, j = 1:2], container = DataFrame)
+    @variable(model, y[i = 2:4, j = 1:2; isodd(i+j)], container = DataFrame)
```

Comment:
The more I think about this, the less I like it. I prefer the explicit "add a new column to an existing dataframe" than the "construct a dataframe from these indices." The latter approach is just going to encourage the same nested loops as the GAMS example.